EsFormatter.py

```diff
@@ -139,18 +174,22 @@ def handle_threads(self, threads, callback, process=False, lastError=None):
             callback(process, lastError)
 
 
+import tempfile
+
 class NodeCall(threading.Thread):
-    def __init__(self, code, path, options, id=0, region=None):
+    def __init__(self, code, path, id=0, region=None):
         self.code = code.encode('utf-8')
         self.region = region
-        exec_path = os.path.join(sublime.packages_path(), "EsFormatter", "lib", "esformatter.js")
-        self.cmd = getNodeCommand(exec_path, path, options)
         self.result = None
         threading.Thread.__init__(self)
 
     def run(self):
         try:
-            process = subprocess.Popen(self.cmd, bufsize=160*len(self.code), stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE, startupinfo=getStartupInfo())
+            f = tempfile.NamedTemporaryFile(delete=False)
+            f.write(self.code)
+            f.close()
```

Comment:
probably the file should be deleted after calling Popen otherwise we keep creating new files